### ISSUE TITLE
AnimeBytes implementation need a bit of touch up. And currently broken.

### ISSUE THREAD

**Issue by Ether009** [Jackett/Jackett#1942]
So first thing is that for AnimeBytes, while searching for season and episodes isn't possible, since that information is available on the page, shouldn't that be possible to filter out all the incorrect ones and present only the episodes matching the search? Season I guess might be harder since that would be a different series at times and such, but then the next thing comes in...

Currently, AnimeBytes searching actually strips away certain things from the search making it a lot broader than it should be... A Search for as an example "Kekkai Sensen & Beyond", results in the search being passed to AnimeBytes being just "Kekkai Sensen", which gives a much broader result with specials and first season and such. AnimeBytes does allow searching with special chars like & so there's no reason to be stripping out that part.

That problem becomes even more glaring with the last issue though that's more of a request perhaps. Please make an option to turn off presenting every single synonym as a unique torrent in the result. In the above example of Kekkai Sensen, the first season has 13 torrents, but when Jackett gives its list, the list is now 13 torrents for Kekkai Sensen, 13 torrents for 血界戦線, and 13 torrents for Blood Blockade Battlefront. This becomes even worse when it's a longer running show with many synonyms... As an example, Welcome to the Ballroom... It so far has 14 episodes... Each episode has 3 torrents (except for last 3 eps which only has 2 so far). Each torrent has 3 duplicates... The result is 39 unique torrents which by itself is a bit of an issue, but with synonyms, it now has a wooping 156 torrents. While I realize ofc that some setups will need those synonyms to work, there's no real reason to force that setting on us poor Sonarr users since those 156 torrents, result in a search that isn't possible, because it overflows and thus, the useful torrents gets ignored.

TLDR: Please fix the clipping of the show name. Please filter out the episodes that wasn't being searched for. And please at least allow for an option to turn off duplicating all the torrents.

**Comment by Ether009** [comment]
Adding that it's currently broken. Test in Jackett gives a Parse Error. Log gives
```
Error on 20171011051338 for AnimeBytes.txtSystem.InvalidOperationException: Sequence contains no elements
   at System.Linq.Enumerable.Last[TSource](IEnumerable`1 source)
   at Jackett.Indexers.AnimeBytes.<GetResults>d__20.MoveNext() in C:\projects\jackett\src\Jackett\Indexers\AnimeBytes.cs:line 371
```

**Comment by qtv** [comment]
and then there is https://github.com/Jackett/Jackett/issues/1535 which is still affecting people (like me).

**Comment by Dhs92** [comment]
Receiving this error as well

**Comment by kaso17** [comment]
63cf687f02828aa7197bd53f98e347ebde3dc2b5 adds options for synonyms and season/episode filtering
searching for "Kekkai Sensen & Beyond" via the Jackett manual search works just fine. I suspect sonarr might be already stripping parts of the title. Can you please check your sonarr/jackett logs to get the exact torznab URL?

I also can't reproduce the Parse Error, can you please provide a full enhanced log for it?

**Comment by Ether009** [comment]
The parse error happens due to something in the new feed. It solved itself when that item left the feed. It does show that there is something that it doesn't like in the formatting of some things that does show up, but even with extended logging, the logging isn't expressive enough to say what.

As for searching for Kekkai Sensen & Beyond working, well the thing here is that manually searching works just fine in Jackett, and looking at the request that Sonarr makes, does include the & Beyond or well (and Beyond actually).

`17-10-17 11:49:49.0|Debug|Torznab|Downloading Feed http://127.0.0.1:9117/torznab/animebytes/api?t=search&cat=5070&extended=1&apikey=(removed)&offset=0&limit=100&q=Kekkai%20Sensen%20and%20Beyond+02`

That being said... I did notice something else when looking through that so I'm glad you brought it up anyway again. Because I notice that's actually wrong since AnimeBytes wouldn't even return anything with that search because AB doesn't treat & and "and" as being the same thing at all. So gonna have to report that error to Sonarr. I also see that there's a second search, for just Kekkai Sensen triggered right after that, so it would seem that Sonarr is indeed the culprit here and sorry for jumping to conclusions. I know I did check that it sent the "and beyond" before making entry, but I missed that it made a second request as well after that.

I would like to thank you for the option for synonyms and season/episode filtering though. Much appreciated. Since the parse error fixed itself, do we assume it's a one off and close or keep open until it happens again and I can perhaps monitor more closely what items are currently on the new page for it and see when it goes off again to see more precisely what item triggered it? (The lines I posted above, really was the extensive logging for the error. It didn't say more).

**Comment by kaso17** [comment]
With enhanced logging enabled it should dump the complete search result HTML page, with that information we can reproduce and fix such errors.
Closing this issue for now.

**Comment by Ether009** [comment]
Oh it did. But only in the logfile, not the UI.
[log.20171011.00067.txt](https://github.com/Jackett/Jackett/files/1390701/log.20171011.00067.txt)

**Comment by kaso17** [comment]
Fixed via bb80da9b197a1cbc675e6d4cac2cc8caeaa0728e

Was caused by https://animebytes.tv/torrents.php?id=47180 which doesn't have any tags.
